src/schema/v2/quiz.ts

```diff
@@ -1,18 +1,39 @@
-import { GraphQLFieldConfig } from "graphql"
+import { GraphQLFieldConfig, GraphQLList } from "graphql"
 import { GraphQLNonNull, GraphQLObjectType } from "graphql"
 import { InternalIDFields } from "schema/v2/object_identification"
 import { quizArtworkConnection } from "./quizArtworkConnection"
 import { ResolverContext } from "types/graphql"
 import { date } from "schema/v2/fields/date"
+import { ArtworkType } from "./artwork"
 
-export const QuizType = new GraphQLObjectType({
+export const QuizType = new GraphQLObjectType<any, ResolverContext>({
   name: "Quiz",
   fields: () => ({
     ...InternalIDFields,
     quizArtworkConnection,
     completedAt: date(
       ({ completed_at }: { completed_at: string | null }) => completed_at
     ),
+    savedArtworks: {
+      type: new GraphQLNonNull(
+        new GraphQLList(new GraphQLNonNull(ArtworkType))
+      ),
+      resolve: async ({ quiz_artworks }, _, { savedArtworkLoader }) => {
+        if (!savedArtworkLoader) return []
+
+        try {
+          const artworks = await Promise.all(
+            quiz_artworks.map(({ artwork_id }) => {
+              return savedArtworkLoader(artwork_id)
+            })
+          )
```

Comment:
This is bundled as a single request because of the dataloader pattern:

![](https://static.damonzucconi.com/_capture/BpdmGtq6.png)

Reply:
That's wild that you get that behavior without being aware of it!!